File: app/src/main/java/za/org/grassroot2/view/GrassrootView.java
package za.org.grassroot2.view;

import android.app.Activity;
import android.os.Bundle;

import za.org.grassroot.messaging.dto.MessageDTO;

public interface GrassrootView extends ProgressBarContainer {


    Activity getActivity();
    void showSuccessMsg(int successMsg);
    void showErrorSnackbar(int errorTextRes);
    void closeKeyboard();
    void launchActivity(Class<?> cls, Bundle args);

    void handleNoConnection();
    void handleNoConnectionUpload();
    void showNoConnectionMessage();

    void showMessageDialog(String text);

    void hideKeyboard();

    void sendCGMMessage(MessageDTO messageDTO);
}

File: app/src/main/java/za/org/grassroot2/view/GrassrootView.kt
package za.org.grassroot2.view

import android.app.Activity
import android.os.Bundle
import android.view.View

interface GrassrootView : ProgressBarContainer {

    val activity: Activity

    fun showSuccessSnackbar(successMsg: Int)
    fun showErrorSnackbar(errorTextRes: Int)
    fun showSuccessDialog(textRes: Int, okayListener: View.OnClickListener)
    fun showErrorDialog(errorMsg: Int)

    fun closeKeyboard()
    fun launchActivity(cls: Class<*>, args: Bundle)

    fun handleNoConnection()
    fun handleNoConnectionUpload()
    fun showNoConnectionMessage()

    fun hideKeyboard()
}